spec/Functions.tex

```diff
@@ -717,24 +717,29 @@ \subsection{The Ref Return Intent}
 
 When using a \chpl{ref} return intent, the function call is an lvalue
 (specifically, a call expression for a procedure and an iterator variable for
-an iterator).  This implies that the function produces a reference; however,
-this reference cannot be captured.
-%TODO: is this still true with ref variables?
+an iterator).
 
 The \chpl{ref} return intent is specified by following the argument list with
 the \chpl{ref} keyword.  The function must return or yield an lvalue.
-% Todo: disallow void return types for variable functions?
 
-When a procedure with a \chpl{ref} return intent is called on the left-hand
-side of an assignment statement or in the context of a call to a formal
-argument by \chpl{out}, \chpl{inout}, or \chpl{ref} intent, the lvalue that is
-returned by the procedure is assigned a value.
+During function resolution, if a procedure with a \chpl{ref} return
+intent and a procedure without a \chpl{ref} return intent are both
+possible candidates, and either neither use promotion or both do,
+this pair fo procedures is called a ref-pair.
 
-Functions with a \chpl{ref} return intent provide an implicit \chpl{param}
-formal argument \chpl{setter} of type \chpl{bool}.  If the function is called
-in a context requiring an lvalue, the actual argument for \chpl{setter} is
-implicitly \chpl{true}; otherwise it is \chpl{false}.  This argument is useful
-for adjusting the behavior depending to the calling context.
+When a procedure in a ref-pair is called on the left-hand side of an
+assignment statement or in the context of a call to a formal argument by
+\chpl{out}, \chpl{inout}, or \chpl{ref} intent, the \chpl{ref} return
+intent procudure is used. Otherwise, the other candidate is used.  This
+behavior can be useful for adjusting the behavior depending to the
+calling context.
+
+\begin{craychapel}
+Previous versions of Chapel included an implicit \chpl{setter} param of
+type \chpl{bool} for \chpl{ref} return intent functions. The old behavior
+can be recreated by making both ref-pair functions call another function
+and explicitly pass the \chpl{setter} value, as in the the example below.
+\end{craychapel}
```

Comment:
Let's go ahead with the above paragraph for now.
After the fact check with Brad if it is appropriate to have it.